Code QL pre-migration tasks: update articles relating to CodeQL for JavaScript

mchammer01 opened the pull request:
See https://github.com/github/product-documentation/issues/2165 for details.
The first commit changes titles.

mchammer01 commented:
2nd commit adds some intros but I'm afraid I won't get much further before we have to leave for the mini-summit.

mchammer01 commented:
Please note that I haven't made any changes to do with content models and localization checklists.

intrigus-lgtm commented on docs/language/learn-ql/javascript/introduce-libraries-js.rst:
```diff
@@ -1,5 +1,7 @@
+CodeQL libraries for JavaScript
+===============================
+
+This article is an overview of the extensive libraries you can use to analyze databases generated from JavaScript codebases. Using thes libraries makes it easier for you to write queries.
```
Typo: "thes libraries"

intrigus-lgtm commented on docs/language/learn-ql/javascript/introduce-libraries-ts.rst:
```diff
@@ -1,5 +1,7 @@
+CodeQL libraries for TypeScript
+===============================
+
+This article is an overview of the libraries you can use to analyze databases generated from TypeScript codebases. Using thes libraries makes it easier for you to write queries.
```
Same typo here.

intrigus-lgtm commented on docs/language/learn-ql/javascript/type-tracking.rst:
```diff
@@ -1,5 +1,5 @@
-Tutorial: API modelling using type tracking
-===========================================
+Using type tracking for API modeling
```
Nitpicking: If the docs don't use American English, this should probably be `modelling` with two `l`s?

felicitymay replied:
Thanks for highlighting this. The docs aim to use American English, so this is correct.

mchammer01 commented:
Thanks for the valuable review @intrigus-lgtm 👍 
This PR has been raised as a draft so I wasn't expecting reviews just yet 😉 
It's just a starting point and I have a lot more work to do but I'll ensure I'll take your comments (especially the ones about the typos 😱 ) into consideration.
Our documentation is meant to be written in US English but I know that we need to rationalise spellings and check that this is true throughout the whole set of articles.
Thanks again for your help!

mchammer01 commented:
This PR is now ready for review.
The work in this PR contains pre-migration tasks for the CodeQL documentation. This includes:
- updating article titles
- adding intros to each articles
- updating links as per the new article title

mchammer01 commented:
Apologies for the automatic review request @jf205 and @shati-patel but I can't seem to be able to remove that (is this a permissions thing on this repo)?

felicitymay commented:
Since this is small text changes, I'm removing @jf205 ad @shati-patel as reviewers. Anyone in the docs team can review this PR.

felicitymay commented on docs/language/learn-ql/javascript/introduce-libraries-ts.rst:
```diff
@@ -10,7 +12,7 @@ Support for analyzing TypeScript code is bundled with the CodeQL libraries for J
    import javascript
 
-The :doc:`CodeQL library introduction for JavaScript <introduce-libraries-js>` covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.
+The :doc:`CodeQL libraries for JavaScript <introduce-libraries-js>` covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.
```
With the change in topic name, it reads a little weirdly now. Suggest dropping "the".

```suggestion
:doc:`CodeQL libraries for JavaScript <introduce-libraries-js>` covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.
```

felicitymay commented on docs/language/learn-ql/javascript/type-tracking.rst:
```diff
@@ -489,7 +489,7 @@ Prefer type tracking when:
 
 - Tracking user-controlled data -- use `taint tracking <https://help.semmle.com/QL/learn-ql/javascript/dataflow.html#using-global-taint-tracking>`__.
-- Differentiating between different kinds of user-controlled data -- use :doc:`flow labels <flow-labels>`.
+- Differentiating between different kinds of user-controlled data -- use :doc:`Using flow labels for precise data flow analysis <flow-labels>`.
```
With the link name change, suggest:
```suggestion
- Differentiating between different kinds of user-controlled data -- see :doc:`Using flow labels for precise data flow analysis <flow-labels>`.
```

mchammer01 commented:
@felicitymay - thanks for your review. I believe I have addressed your comments.
I am keen to get this merged today, as discussed in our team meeting. 
We can still iterate if there are further things to tweak or rationalize between languages.

felicitymay commented on docs/language/learn-ql/javascript/type-tracking.rst:
```diff
@@ -1,9 +1,11 @@
+====================================
 
-This tutorial demonstrates how to build a simple model of the Firebase API
+You can build a simple model of the Firebase API
```
It looks as if Firebase is just used as an example, so suggest making this a little more general:

```suggestion
You can track data through an API by creating a model 
```

felicitymay commented on docs/language/learn-ql/javascript/dataflow-cheat-sheet.rst:
```diff
@@ -1,5 +1,5 @@
+Data flow cheat sheet for JavaScript
+====================================
 
 This page describes parts of the JavaScript libraries commonly used for variant analysis and in data flow queries.
```
```suggestion
This article describes parts of the JavaScript libraries commonly used for variant analysis and in data flow queries.
```

mchammer01 commented:
@felicitymay - I have accepted your suggestions. Can you double-check you are ok with things now? Thanks.

felicitymay commented:
All LGTM 💚
